cmd/format-disk-cache.go

```diff
@@ -116,62 +122,41 @@ func createFormatCache(fsFormatPath string, format *formatCacheV1) error {
 // of format cache config
 func initFormatCache(drives []string) (formats []*formatCacheV1, err error) {
 	nformats := newFormatCacheV1(drives)
-	for i, drive := range drives {
-		// Disallow relative paths, figure out absolute paths.
-		cfsPath, err := filepath.Abs(drive)
-		if err != nil {
+	for _, drive := range drives {
+		if _, err := os.Stat(drive); err != nil {
+			// During init cache, all drives have to be available.
 			return nil, err
 		}
-
-		fi, err := os.Stat(cfsPath)
-		if err == nil {
-			if !fi.IsDir() {
-				return nil, syscall.ENOTDIR
-			}
-		}
-		if os.IsNotExist(err) {
-			// Disk not found create it.
-			err = os.MkdirAll(cfsPath, 0777)
-			if err != nil {
-				return nil, err
-			}
+	}
+	for i, drive := range drives {
+		if err := os.Mkdir(pathJoin(drive, minioMetaBucket), 0777); err != nil {
+			return nil, err
 		}
-
-		cacheFormatPath := pathJoin(drive, formatConfigFile)
+		cacheFormatPath := pathJoin(drive, minioMetaBucket, formatConfigFile)
 		// Fresh disk - create format.json for this cfs
-		if err = createFormatCache(cacheFormatPath, nformats[i]); err != nil {
+		if err := createFormatCache(cacheFormatPath, nformats[i]); err != nil {
 			return nil, err
 		}
 	}
 	return nformats, nil
 }
 
-func loadFormatCache(drives []string) (formats []*formatCacheV1, err error) {
-	var errs []error
-	for _, drive := range drives {
-		cacheFormatPath := pathJoin(drive, formatConfigFile)
+func loadFormatCache(drives []string) []*formatCacheV1 {
+	formats := make([]*formatCacheV1, len(drives))
+	for i, drive := range drives {
+		cacheFormatPath := pathJoin(drive, minioMetaBucket, formatConfigFile)
 		f, perr := os.Open(cacheFormatPath)
 		if perr != nil {
-			formats = append(formats, nil)
```

Comment:
shouldn't we log errors for these?

Reply:
no, because it is the cache disks have been unmounted because of disk failure.